### DISCUSSION TITLE
Pointer > Disable pointer acceleration Functionality

### DISCUSSION THREAD

**Discussion by akfranco** [linearmouse/linearmouse#201] [Q&A] [answered]
I am trying to understand the difference between manually selecting an `Acceleration` of `0` and selecting the `Disable Pointer Acceleration` option. Any help would be appreciated. Thank you.

**Comment by lujjjh** [comment, marked as answer]
I prefer to manually set acceleration to 0 and tune the pointer speed (aka pointer sensitivity in some other apps) as well. But for many people, one-click "disable pointer acceleration" is also a good option.

~Under the hood, "disable pointer acceleration" sets acceleration to -1, which seems to be a magic number in macOS that disables pointer acceleration with a reasonable pointer speed rather than a very slow pointer speed.~

> ~**Warning**
> There is a [known issue](https://github.com/linearmouse/linearmouse/issues/106) of "disable pointer acceleration" in universal control.
> I'm not sure if it has been fixed in recent macOS releases.~

Update: Beginning with macOS Sonoma, LinearMouse now utilizes an official method to disable pointer acceleration that functions the same as the setting in the System Settings.

My suggestion is if you prefer to manually adjust the pointer speed like I do, you may set acceleration to 0. Otherwise, you can simply toggle "disable pointer acceleration" on.

**Reply by ciriousjoker** [reply]
A couple of follow up questions:
- Am I correct in interpreting this as "both ways are identical in the end result if you set the right speed value"? Setting pointer acceleration to 0 doesn't seem to _actually_ disable it as the cursor will still move more distance if it's faster.
- What is the right speed value? Ie what speed is mathematically identical to just enabling the toggle? 0.42 seems to be pretty good but is it mathematically correct?
- The speed scale doesn't seem to be linear, ie 0.4 isn't half as fast as 0.8

**Reply by lujjjh** [reply]
1. There seems to be no difference from my side. I did a rough test with my ERGO M575.
2. I believe this value can be calculated with [the acceleration table](https://opensource.apple.com/source/IOHIDFamily/IOHIDFamily-1633.81.2/IOHIDSystem/IOHIPointing.cpp.auto.html), but I haven't dived into it.
3. This may be related to the first question. Have you tried setting the acceleration to zero and moving the mouse at the same speed?

**Reply by ciriousjoker** [reply]
1. I put a stationary object (my hand) on my mousepad. When moving the mouse to the right I move it slow, when going left and mashing it against my hand, I'm going fast. After a couple of times, the cursor is further to the left even though my hand still stops the mouse at the same location.
2. No clue how to use this, but I only skimmed through it atm since I'm on my phone
3. Is the scale supposed to be linear? Maybe my eyes are just broken here. Yes acceleration is 0

**Reply by lujjjh** [reply]
The pointer speed setting is supposed to be linear. The underlying property is called `HIDPointerResolution` and if the property value is halved, the pointer should move twice as fast as its original speed, just like the display resolution is reduced. The pointer speed setting is inversely proportional to `HIDPointerResolution`, so it should be linear. For the formula, please refer to https://github.com/linearmouse/linearmouse/issues/65#issuecomment-1173500052 where $\tt{sensitivity}$ is `HIDPointerResolution`.

I'm not sure why the pointer acceleration still exist when you set it to zero. Have you tested whether the "disable pointer acceleration" option is functioning properly?

**Reply by ciriousjoker** [reply]
Few cases I tested:
1. "disable pointer acceleration": works pixel perfect
2.  "acceleration 0 + speed 0.4": works pixel perfect (at least I can't tell a difference between this and the first)
3. "acceleration 0 + speed 0.8": I've managed to move the cursor from the right to the left of the screen by repeatedly mashing it against my stationary fingers (ie go slow when to the right, speed up towards the stationary object). Stationary object didn't move, but cursor went past the window border I used to mark the starting location (and eventually ended up on the left of the screen after 15 times or so).

**Reply by cakiya** [reply]
currently on ventura, is there a way to set acceleration to -1 ("Disable pointer acceleration") without the linearmouse program? i've tried using the terminal command

defaults write -g com.apple.mouse.scaling -1

but this doesn't work; after rebooting, moving my cursor around with my touchpad feels off (acceleration is still there) and opening linearmouse changes the movement of my cursor to actually have no acceleration.

could you explain what exactly goes on under the hood for the "Disable pointer acceleration" option in linearmouse for ventura? I'd like to cut down on startup processes as much as possible and since i only use linearmouse for this one option, if i can get it without linearmouse that would be great.

**Reply by lujjjh** [reply]
@cakiya Are you attempting to disable pointer acceleration for your mouse or trackpad? This command will not affect trackpads.

If you're looking to disable pointer acceleration for your trackpad, use the following command instead:

```sh
defaults write -g com.apple.trackpad.scaling -1
```

**Reply by cakiya** [reply]
this works, thank you so much! <3

**Comment by lukerucker** [comment]
From what I can tell in subjective testing, 'Disable pointer acceleration' a la MacOS acceleration '-1' corresponds to Acceleration: 0 and Speed: 0.35. This is probably the recommended native scaling to Windows raw input, but this may or may not be the case.

**Reply by skareel** [reply]
Are we able to test this out against windows? I want to match the mouse feel exactly between windows and mac.

**Reply by we-taper** [reply]
This can be quite complicated since the Windows' acceleration curve seems to be a [complex composition of several logarithmic functions](https://www.reddit.com/r/MouseAccel/comments/jxizqz/windows_10_mouse_accel_curve/).

**Comment by hyperknot** [comment]
How is LinearMouse's Pointer: Speed relate to System Preferences / Mouse / Tracking speed?

Is it the same setting? Is LinearMouse overwriting the SysPref one?

**Reply by lujjjh** [reply]
The tracking speed in System Preferences actually refers to the pointer acceleration.

macOS does not provide a direct way to modify the pointer speed (i.e. pointer sensitivity) and that's why apps like LinearMouse and CursorSense exist.

**Reply by hyperknot** [reply]
OK, so if I put 0 to acceleration then I can disregard the macOS plan?

2nd question, how do I convert between the 0-1990 sensitivity of CursorSense to LinearMouse's 0-1 scale? I was using 1900 in CursorSense (which I bought but I much prefer LinearMouse, thanks for developing it!)

**Reply by lujjjh** [reply]
It's about 0.3793. You can find the formula [here](https://github.com/linearmouse/linearmouse/issues/65#issuecomment-1173500052).

**Reply by hyperknot** [reply]
Thanks a lot!

**Comment by caldempsey** [comment]
Hi @lujjjh ,

Curious why disabling the mouse acceleration in MacOSX via the command line doesn't work produce the same result as Linear Mouse's disable mouse acceleration function.

Seems like Linear Mouse removes the acceleration as expected, unlike the Command Line `defaults write .GlobalPreferences com.apple.mouse.scaling -1`

Would like to configure this for my dot files without installing Linear Mouse, can you help with this?

**Reply by lujjjh** [reply]
This command requires a logout / reboot to take effect.

**Reply by caldempsey** [reply]
Have you turned turning it on and off again?

Classic!

**Comment by Brownagedon** [comment]
It's probably just be placebo, but I swear I can feel a difference between the disable accel option and setting the accel to 0.

**Reply by caldempsey** [reply]
see my comment
